From fedd0787834377f88d6c5a8c39453c5e11e0a5e6 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Wed, 5 Apr 2023 20:17:54 +0300
Subject: [PATCH] CMake: Make coverage build truly optional

This is facilitated by UBports cmake-extras[1], however without it
around during CMake configure stage even when not doing a coverage build
this will result in the following fatal error:

  CMake Error at tests/unit/CMakeLists.txt:36 (enable_coverage_report):
    Unknown CMake command "enable_coverage_report".

[1] https://gitlab.com/ubports/development/core/cmake-extras/-/blob/main/src/CoverageReport/CoverageReportConfig.cmake
---
 CMakeLists.txt            |  4 +++-
 tests/unit/CMakeLists.txt | 21 +++++++++++----------
 2 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index add8eb4..984f49e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,4 +45,6 @@ endif()
 #  * Find html report in subdir coveragereport
 #  * Find xml report feasible for jenkins in coverage.xml
 #####################################################################
-find_package(CoverageReport)
+if(cmake_build_type_lower MATCHES coverage)
+    find_package(CoverageReport)
+endif()
diff --git a/tests/unit/CMakeLists.txt b/tests/unit/CMakeLists.txt
index 34838a2..a27b7ac 100644
--- a/tests/unit/CMakeLists.txt
+++ b/tests/unit/CMakeLists.txt
@@ -35,13 +35,14 @@ declare_test(TestConfig tst_config)
 declare_test(TestDeviceinfo tst_deviceinfo)
 declare_test(TestUtils tst_utils)
 
-find_package(CoverageReport)
-enable_coverage_report(
-  TARGETS
-    deviceinfo
-  FILTER
-    ${CMAKE_SOURCE_DIR}/tests/*
-    ${CMAKE_BINARY_DIR}/*
-  TESTS
-    ${TESTS}
-)
+if(cmake_build_type_lower MATCHES coverage)
+  enable_coverage_report(
+    TARGETS
+      deviceinfo
+    FILTER
+      ${CMAKE_SOURCE_DIR}/tests/*
+      ${CMAKE_BINARY_DIR}/*
+    TESTS
+      ${TESTS}
+  )
+endif()
-- 
GitLab
